tools/notebook-status.sh

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env bash
```

Comment:
This would have been much better in python but NBD :)

Reply:
The GIL was holding this script back

Reply:
GIL?

Are you running everything in parallel in bash? 

C++ would be better ;)